sqlglot/dialects/duckdb.py

```diff
@@ -867,3 +869,18 @@ def generatedatearray_sql(self, expression: exp.GenerateDateArray) -> str:
 
             # The result is TIMESTAMP array, so to match BQ's semantics we must cast it back to DATE array
             return self.sql(exp.cast(gen_series, exp.DataType.build("ARRAY<DATE>")))
+
+        def generatetimestamparray_sql(self, expression: exp.GenerateDateArray) -> str:
+            start = _implicit_datetime_cast(
+                expression.args.get("start"), type=exp.DataType.Type.TIMESTAMP
+            )
+            end = _implicit_datetime_cast(
+                expression.args.get("end"), type=exp.DataType.Type.TIMESTAMP
+            )
+
+            # BQ's GENERATE_TIMESTAMP_ARRAY is transformed to DuckDB'S GENERATE_SERIES
+            gen_series = exp.GenerateSeries(
+                start=start, end=end, step=expression.args.get("interval")
+            )
+
+            return self.sql(gen_series)
```

Comment:
Don't we need to cast to `ARRAY<TIMESTAMP>` here, similar to `generatedatearray_sql`? It looks like these two methods can be consolidated, the logic is almost the same. I'd consider making it more DRY.

Reply:
That was needed in `GENERATE_DATE_ARRAY` because DDB'S `GENERATE_SERIES` outputs a timestamp array by default.

I also thought of combining the logic but it will still require branching logic to implement the different casts etc. Can give it a shot though to see how it'd look